addons/event_sale/models/sale_order.py

```diff
@@ -1,44 +1,32 @@
 # -*- coding: utf-8 -*-
 
-from openerp import api
-from openerp.osv import fields, osv
+from openerp import api, fields, models
 
 
-class sale_order(osv.osv):
+class SaleOrder(models.Model):
     _inherit = "sale.order"
 
-    def action_confirm(self, cr, uid, ids, context=None):
-        res = super(sale_order, self).action_confirm(cr, uid, ids, context=context)
-        for order in self.browse(cr, uid, ids, context=context):
-            redirect_to_event_registration, so_id = any(line.event_id for line in order.order_line), order.id
-            order.order_line._update_registrations(confirm=True)
-        if redirect_to_event_registration:
-            event_ctx = dict(context, default_sale_order_id=so_id)
-            return self.pool.get('ir.actions.act_window').for_xml_id(cr, uid, 'event_sale', 'action_sale_order_event_registration', event_ctx)
-        else:
-            return res
+    @api.multi
+    def action_confirm(self):
+        res = super(SaleOrder, self).action_confirm()
+        if self.mapped('order_line.event_id'):
+            return self.env['ir.actions.act_window'].with_context(default_sale_order_id=self.id).for_xml_id('event_sale', 'action_sale_order_event_registration')
+        return res
 
 
-class sale_order_line(osv.osv):
+class SaleOrderLine(models.Model):
     _inherit = 'sale.order.line'
-    _columns = {
-        'event_id': fields.many2one(
-            'event.event', 'Event',
-            help="Choose an event and it will automatically create a registration for this event."),
-        'event_ticket_id': fields.many2one(
-            'event.event.ticket', 'Event Ticket',
-            help="Choose an event ticket and it will automatically create a registration for this event ticket."),
-        # those 2 fields are used for dynamic domains and filled by onchange
-        # TDE: really necessary ? ...
-        'event_type_id': fields.related('product_id', 'event_type_id', type='many2one', relation="event.type", string="Event Type"),
-        'event_ok': fields.related('product_id', 'event_ok', string='event_ok', type='boolean'),
-    }
 
-    def _prepare_order_line_invoice_line(self, cr, uid, line, account_id=False, context=None):
-        res = super(sale_order_line, self)._prepare_order_line_invoice_line(cr, uid, line, account_id=account_id, context=context)
+    event_id = fields.Many2one('event.event', 'Event', help="Choose an event and it will automatically create a registration for this event.")
+    event_ticket_id = fields.Many2one('event.event.ticket', 'Event Ticket', help="Choose an event ticket and it will automatically create a registration for this event ticket.")
+    event_type_id = fields.Many2one(related='product_id.event_type_id', string="Event Type")
+    event_ok = fields.Boolean(related='product_id.event_ok')
+
+    @api.model
+    def _prepare_order_line_invoice_line(self, line, account_id=False, context=None):
```

Comment:
This method seems been removed from main app and I think it is replaced by _prepare_invoice_line(...), would you please check it and override _prepare_invoice_line here.